### PR TITLE
Pm 1465 fix chiamate asincrone vpos xpay

### PR DESCRIPTION
**Pull request by robertorasi-nexigroup**
#### List of Changes

Fix asynchronous behaviour moving method marked as @Async in dedicated classes and changing scope to public.

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have created or updated an API interface.
  - [ ] I have updated the OpenAPI spec
- [ ] I have created or updated a database table
  - [ ] I have updated both the Oracle and Postgres migrations
- [ ] My PR is cohesive and self-contained (rule of thumb: less than 250 changed lines)

### CODE REVIEW

**Comment by giovanniberti** [comment]
Given that this PR touches many different parts of the payment process, can you split this into many different PRs? (e.g. one for each endpoint)

File: src/main/java/it/pagopa/pm/gateway/service/async/CcResumeStep1AsyncService.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    private VPosRequestUtils vPosRequestUtils;
+
+    @Autowired
+    private VPosResponseUtils vPosResponseUtils;
+
+    @Autowired
+    private HttpClient httpClient;
+
+    @Autowired
+    private EcommercePatchUtils vposPatchUtils;
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/pm/gateway/service/async/CcResumeStep1AsyncService.java]: Prefer constructor injection to field injection

File: src/main/java/it/pagopa/pm/gateway/service/async/VposAsyncService.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+                String authNumber = ((ThreeDS2Authorization) response.getThreeDS2ResponseElement()).getAuthorizationNumber();
+                executeAccount(entity, pgsRequest, authNumber);
+            }
+
+            //If the resultCode is 25 or 26, the PATCH is not called
+            if (!METHOD_CHALLENGE_CODES.contains(response.getResultCode())) {
+                ecommercePatchUtils.executePatchTransaction(entity);
+            }
+        } catch (Exception e) {
+            log.error(GENERIC_ERROR_MSG + entity.getIdTransaction() + CAUSE + e.getCause() + " - " + e.getMessage(), e);
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/pm/gateway/service/async/VposAsyncService.java:74]:
```suggestion
            log.error("{}{}{} - {}", GENERIC_ERROR_MSG , entity.getIdTransaction() , CAUSE , e.getCause() , " - " , e.getMessage(), e);
```

**Review by giovanniberti — COMMENTED** [review]
Can you move tests for the async calls into a `async` package in test sources to mirror how it is in production code?

File: src/main/java/it/pagopa/pm/gateway/service/async/VposAsyncService.java (hunk 2 above)
Comment (robertorasi-nexigroup) [reply src/main/java/it/pagopa/pm/gateway/service/async/VposAsyncService.java]: The suggestion is missing some placeholders.
Committed a corrected version on branch.

File: src/main/java/it/pagopa/pm/gateway/service/async/CcResumeStep1AsyncService.java (hunk 1 above)
Comment (robertorasi-nexigroup) [reply src/main/java/it/pagopa/pm/gateway/service/async/CcResumeStep1AsyncService.java]: changed

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**